packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/task/s3_upload_task.dart

```diff
@@ -574,6 +594,35 @@ class S3UploadTask {
     }
   }
 
+  Future<void> _handleUploadPart({
+    required int partNumber,
+    required Future<_CompletedSubtask> uploadPartRequest,
+  }) {
+    return uploadPartRequest
+        .then(_subtasksStreamController.add)
+        .catchError((Object error) {
+      if (error is CancellationException) {
+        _logger
+            .debug('Part $partNumber upload HTTP operation has been canceled.');
+        return;
+      }
+
+      _terminateMultipartUploadOnError(error, isCancel: false);
+    });
+  }
+
+  Future<void> _cancelOngoingUploadPartOperations({
+    bool cancelingOnPause = false,
+  }) async {
+    for (final operation in _ongoingUploadPartHttpOperations) {
+      await operation.smithyOperation.cancel();
+      if (!cancelingOnPause) {
+        _ongoingSubtasks
```

Comment:
Should this be `Map<int, _OngoingSubtask>` so these list searches aren't needed?

Reply:
Applies to other lists as well